march_hardware/src/IMotionCube.cpp

```diff
@@ -61,14 +65,14 @@ void IMotionCube::validateMosiPDOs()
 }
 
 // Set configuration parameters to the IMC
-void IMotionCube::writeInitialSettings(uint8 ecatCycleTime)
+void IMotionCube::writeInitialSettings(uint8 ecatCycleTime, uint8_t modeofOp)
 {
   ROS_DEBUG("IMotionCube::writeInitialSettings");
   bool success = true;
   // sdo_bit32(slaveIndex, address, subindex, value);
   // mode of operation
-  success &= sdo_bit8(slaveIndex, 0x6060, 0, 8);
-
+  ROS_INFO("the mode of operation is: %i", modeofOp);
```

Comment:
If you want to keep this as an info print, please add the joint name (e.g. "Mode of operation for left_knee is torque"). Since the IMC doesn't have the joint name, you probably need to move the print to the joint class